pkg/monitor/monitor/util.go

```diff
@@ -681,6 +681,13 @@ func getMonitorService(monitor *v1alpha1.TidbMonitor) []*core.Service {
 			},
 		},
 	}
+	if monitor.Spec.Prometheus.Service.PortName != nil {
+		prometheusService.Spec.Ports[0].Name = *monitor.Spec.Prometheus.Service.PortName
```

Comment:
Please assign the port name to the specific port object instead of to an index

Reply:
update.